x-pack/plugins/apm/public/components/app/mobile/service_overview/most_used_charts/sunburst_chart.tsx

```diff
@@ -0,0 +1,158 @@
+    idealFontSizeJump: 1.1,
+    outerSizeRatio: 1,
+    emptySizeRatio: 0.3,
+    circlePadding: 3,
+  },
+};
+
+export function SunburstChart({
+  data,
+  label,
+  chartKey,
+  fetchStatus,
+  chartWidth,
+}: {
+  data?: Array<{ key: string | number; docCount: number }>;
+  label?: string;
+  chartKey: string;
+  fetchStatus: FETCH_STATUS;
+  chartWidth: number;
+}) {
+  const colors = euiPaletteColorBlindBehindText({ sortBy: 'natural' });
+  const isDataAvailable = data && data.length > 0;
+  const isLoading = fetchStatus === FETCH_STATUS.LOADING;
+
+  // The loader needs to be wrapped inside a div with fixed height to avoid layout shift
+  const ProgressLoader = (
+    <div style={{ height: '5px' }}>
+      {isLoading && (
+        <EuiProgress
+          size="xs"
+          color="accent"
+          style={{ background: 'transparent' }}
+        />
+      )}
+    </div>
+  );
+
+  return (
+    <EuiFlexItem
+      grow={true}
+      key={chartKey}
+      style={{
+        height: '200px',
+        width: chartWidth,
+      }}
+    >
+      <EuiTitle size="xs">
+        <h2 style={{ fontSize: '0.8571rem' }}>{label}</h2>
+      </EuiTitle>
+      {ProgressLoader}
+      <EuiSpacer size="m" />
+      <ChartContainer
+        hasData={Boolean(isDataAvailable)}
+        status={fetchStatus}
+        height={200}
+        id={`mostUsedChart-${chartKey}`}
+      >
+        {isDataAvailable ? (
+          <Chart>
+            <Settings theme={theme} />
+            <Partition
+              id={chartKey}
+              data={data}
+              layout={PartitionLayout.sunburst}
+              valueAccessor={(d: Datum) => Number(d.docCount)}
+              valueGetter="percent"
+              layers={[
+                {
+                  groupByRollup: (d: Datum) => d.key,
+                  nodeLabel: (d: Datum) => d,
+                  fillLabel: {
+                    fontWeight: 100,
+                    maximizeFontSize: true,
+                    valueFont: {
+                      fontWeight: 900,
+                    },
+                  },
+                  shape: {
+                    fillColor: (_, sortIndex) => {
+                      return colors[sortIndex];
+                    },
+                  },
+                },
+              ]}
+            />
+          </Chart>
+        ) : (
+          <NoResultsFound />
+        )}
+      </ChartContainer>
+    </EuiFlexItem>
+  );
+}
+
+export function NoResultsFound() {
+  return (
+    <div
+      style={{
+        height: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+      }}
+    >
+      <EuiText
+        data-test-subj="mostUsedNoResultsFound"
+        textAlign="center"
+        color="subdued"
+        size="xs"
+      >
+        <EuiIcon type={IconChartDonut} color="subdued" size="l" />
+        <EuiSpacer size="s" />
+        <p>
+          <FormattedMessage
```

Comment:
Nit: i18n.translate is mostly used for this and `FormattedMessage` when there are variables (for consistency).